Tests/ApolloCodegenTests/GlobTests.swift

```diff
@@ -0,0 +1,359 @@
+import XCTest
+import Nimble
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class GlobTests: XCTestCase {
+  let baseURL = CodegenTestHelper.outputFolderURL().appendingPathComponent("Glob")
+
+  // MARK: Setup
+
+  override func setUpWithError() throws {
+    let fileManager = FileManager.default.apollo
+
+    // <outputFolder>/Glob/
+    try fileManager.createDirectoryIfNeeded(atPath: baseURL.path)
+
+    let files = [
+      self.baseURL.appendingPathComponent("file.one").path, // <outputFolder>/Glob/file.one
+      self.baseURL.appendingPathComponent("file.two").path, // <outputFolder>/Glob/file.two
+      self.baseURL.appendingPathComponent("a/file.one").path, // <outputFolder>/Glob/a/file.one
+      self.baseURL.appendingPathComponent("a/b/file.one").path, // <outputFolder>/Glob/a/b/file.one
+      self.baseURL.appendingPathComponent("a/b/c/file.one").path, // <outputFolder>/Glob/a/b/c/file.one
+      self.baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path, // <outputFolder>/Glob/a/b/c/d/e/f/file.one
+      self.baseURL.appendingPathComponent("a/b/c/d/e/f/file.two").path, // <outputFolder>/Glob/a/b/c/d/e/f/file.two
+      self.baseURL.appendingPathComponent("other/file.one").path, // <outputFolder>/Glob/other/file.one
+      self.baseURL.appendingPathComponent("other/file.oye").path // <outputFolder>/Glob/other/file.oye
+    ]
+
+    for file in files {
+      expect(try fileManager.createFile(atPath: file)).to(beTrue())
+    }
+  }
+
+  override func tearDownWithError() throws {
+    try FileManager.default.apollo.deleteDirectory(atPath: baseURL.path)
```

Comment:
does this also need to call `super` after doing this?

Reply:
Probably better to do so. Doubt it will actually be a problem, but not calling super might cause memory leaks or something?

Reply:
I don't know about memory leaks but generally if you're overriding something and you're not explicitly trying to avoid the effects of what you're overridden, it's a good idea to call `super` in case something in the `super` implementation changes.

Reply:
Yup, I'll add the call to `super`. An oversight on my part.